pex/resolve/lock_resolver.py

```diff
@@ -98,8 +101,9 @@ def __init__(
         self._network_configuration = network_configuration
         self._password_entries = password_entries
         self._cache = cache
-        self._use_pep517 = use_pep517
-        self._build_isolation = build_isolation
+        self._build_configuration = attr.evolve(
+            build_configuration, allow_wheels=False, prefer_older_binary=False
```

Comment:
This is a bug. The incoming `BuildConfiguration` from the user may say `allow_builds=False, allow_wheels=True` (the known useful example being a user constraining a `--style universal` lock to just wheels), in which case just flipping `allow_wheels=False` here leads immediately to `BuildConfiguration` failing a pre-condition check (`allow_wheels=False, allow_builds=False` makes no sense and is illegal). The `VCSArtifactDownloadManager` should complete its thought and say it uses the user `BuildConfiguration` but trumps `allow_wheels=False, allow_builds=True` since the `VCSArtifactDownloadManager` must be able to build a wheel from the VCS checkout to do anything useful at all.